### PR TITLE
feat: Add BubbleXY

### PR DESCRIPTION
**Pull request by jbrundage**
@GordonSmith please review

If the pointSizeColumn is null then BubbleXY defaults it to the last column.

### CODE REVIEW

**Comment by jbrundage** [comment]
image

**Comment by jbrundage** [comment]
@GordonSmith It looks like `test/esp/ecl/Workunit` is causing Travis CI to fail

File: packages/chart/src/Scatter.ts [hunk 1]
```diff
@@ -103,16 +103,27 @@ export class Scatter extends XYAxis {
...
         }
-
+        const usePointSizeColumn = this.pointSizeColumn() !== null;
+        let pointSizeColumnIdx;
+        let pointMinValue;
+        let pointMaxValue;
+        if (usePointSizeColumn) {
+            pointSizeColumnIdx = this.columns().indexOf(this.pointSizeColumn());
+            pointMinValue = Math.min.apply(this, this.data().map(n => n[pointSizeColumnIdx]));
+            pointMaxValue = Math.max.apply(this, this.data().map(n => n[pointSizeColumnIdx]));
```
Comment (GordonSmith) [inline packages/chart/src/Scatter.ts]: For consistency you should use "d3 extent":
```typescript
import { extent as d3Extent } from "d3-array";
...
        const extent = d3Extent(data, d=> d[pointSizeColumnIdx]);
```

File: packages/chart/src/Scatter.ts [hunk 2]
```diff
@@ -103,16 +103,27 @@ export class Scatter extends XYAxis {
...
             d.column = layerColumns[d.colIdx];
             d.row = layerData[d.rowIdx];
-            return d;
-        });
 
+            const valRatio = (d.row[pointSizeColumnIdx] - pointMinValue) / (pointMaxValue - pointMinValue);
+            d.size = usePointSizeColumn ? context.minPointSize() + (valRatio * (context.maxPointSize() - context.minPointSize())) : context.pointSize();
+            return d.column === context.pointSizeColumn() ? false : d;
+        }).filter(d => d);
+        console.log("flatData", flatData);
```
Comment (GordonSmith) [inline packages/chart/src/Scatter.ts]: Remove console.log

File: packages/chart/src/Scatter.ts [hunk 3]
```diff
@@ -103,16 +103,27 @@ export class Scatter extends XYAxis {
...
             d.shape = mapShape(context.pointShape());
             d.column = layerColumns[d.colIdx];
             d.row = layerData[d.rowIdx];
-            return d;
-        });
 
+            const valRatio = (d.row[pointSizeColumnIdx] - pointMinValue) / (pointMaxValue - pointMinValue);
+            d.size = usePointSizeColumn ? context.minPointSize() + (valRatio * (context.maxPointSize() - context.minPointSize())) : context.pointSize();
+            return d.column === context.pointSizeColumn() ? false : d;
+        }).filter(d => d);
```
Comment (GordonSmith) [inline packages/chart/src/Scatter.ts]: What is this filter doing?

File: packages/chart/src/Scatter.ts [hunk 4]
```diff
@@ -103,16 +103,27 @@ export class Scatter extends XYAxis {
...
         const layerData = this.layerData(host);
         const flatData = this.flattenData(layerColumns, layerData).map(function (d) {
             d.shape = mapShape(context.pointShape());
             d.column = layerColumns[d.colIdx];
             d.row = layerData[d.rowIdx];
-            return d;
-        });
 
+            const valRatio = (d.row[pointSizeColumnIdx] - pointMinValue) / (pointMaxValue - pointMinValue);
+            d.size = usePointSizeColumn ? context.minPointSize() + (valRatio * (context.maxPointSize() - context.minPointSize())) : context.pointSize();
```
Comment (GordonSmith) [inline packages/chart/src/Scatter.ts]: I thought you were going to use d3-scale for this (you can set the domain to be the "extent" and the range to be min/max point size...

File: packages/chart/src/Scatter.ts [hunk 5]
```diff
@@ -139,38 +150,39 @@ export class Scatter extends XYAxis {
             .attr("class", d => "point series series-" + this.cssTag(d.column))
             .each(function (d2) {
                 const elementSelection = d3Select(this).select(".pointSelection");
+                const pointSize = d2.size;
```
Comment (GordonSmith) [inline packages/chart/src/Scatter.ts]: Minor:  I don't see a need for "pointSize", just use d2.size

File: packages/chart/src/Scatter.ts [hunk 6]
```diff
@@ -65,11 +67,24 @@ export class Scatter extends XYAxis {
...
+                return d3ScaleSqrt();
+        }
+    }
+
     layerEnter(host: XYAxis, element: d3SelectionType, duration: number = 250) {
         super.layerEnter(host, element, duration);
         const context = this;
         this
-            .tooltipHTML(function (d) {
+            .tooltipHTML(function(d) {
```
Comment (GordonSmith) [inline packages/chart/src/Scatter.ts]: I see this white space change in several places in this file?

**Pull request merged**